src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -26,6 +27,58 @@ object WdlExpression {
 
   val unaryOperators = Set("LogicalNot", "UnaryPlus", "UnaryNegation")
 
+  val preEvaluableFunctions_SingleParameter: Seq[String] = Seq("read_int", "read_string")
+
+  /**
+   * Look within the expression for filenames which aren't explicitly listed as outputs. 
+   */
+  def preevaluateExpressionForFilenames(ast: AstNode, lookup: ScopedLookupFunction, functions: WdlFunctions): Try[Seq[WdlFile]] = {
+    ast match {
+      // This is the only case which actually pre-evaluates anything. The other cases are just buck-passers:
+      case a: Ast if a.getName == "FunctionCall"
+        && a.getAttribute("params").asInstanceOf[AstList].asScala.toVector.size == 1
+        && preEvaluableFunctions_SingleParameter.contains(a.getAttribute("name").asInstanceOf[Terminal].getSourceString)
+      => {
+        // Test the pre-evaluation would be valid by using dummy functions:
+        val innerExpression = a.getAttribute("params").asInstanceOf[AstList].asScala.toVector.head
+        val dummyEvaluation = evaluate(innerExpression, lookup, new DummyPreEvaluationFunctions())
+        dummyEvaluation match {
+          case Failure(reason) => Failure(reason)
+          case Success(_) => {
+            // We aren't going to evaluate anything invalid, so let's do this live:
+            evaluate(innerExpression, lookup, functions) match {
+              case Success(value) => Success(Seq(WdlFile(value.valueString)))
+              case Failure(error) => Failure(error)
+            }
+          }
+        }
+      }
+
+      // Binary operators - find filenames in sub-expressions and merge the lists:
```

Comment:
If I'm not mistaken, the code below should take the expression `"foo" + ".txt"` and return `Seq(WdlFile("foo"), WdlFile(".txt"))`?